techmatters/hrm
# [Client Profiles] HRM Migrations and save automation

## Description
This PR:
- Creates the tables needed for Client Profiles feature, and populates them accordingly.
- Adds logic to contact creation to relate client profiles:
  - Create a new profile and identifier if there is no existing record for a given `number`.
  - Reuse the existing ones if the `number` is already saved.

The tables structure is as in the following diagram.
![Screenshot from 2023-10-05 13-09-29](https://github.com/techmatters/hrm/assets/15805319/b19b6493-6f0d-4614-b503-9a11071fe034)


For further details is refer to the below linked ticket.

### Checklist
- [x] [Corresponding issue has been opened](https://tech-matters.atlassian.net/browse/CHI-1908)

### Verification steps
- Start hrm server from `master` branch, and save a few "live" contacts and a few "offline contacts".
- Stop server and start again in this branch.
- Confirm that the migration is successful:
  - Tables are created.
  - `Identifiers` and `Profiles` tables are populated with the present `number`s in the `Contacts` table.
  - `Contacts` table is updated: `profileId` and `identifierId` are populated ONLY for those records with `number` (no offline contacts).
- Save new contacts and confirm the existing identifiers are re-used.
- Use a new identifier (or delete one of the existing ones) and save a new contact, confirm that the new profile and identifier are saved before saving the contact.
